keps/sig-apps/3998-job-success-completion-policy/README.md

```diff
@@ -0,0 +1,779 @@
+	// "succeededIndexes" or "succeededCount".
+	// These criteria are evaluated in order, and the job meets one of the criteria,
+	// other criteria are ignored.
+	// After the job meets one of the criteria, the remaining running pods are removed.
+	//
+	// +optional
+	Criteria []SuccessPolicyCriterion
+}
+
+// SuccessPolicyCriteria describes a Job can be succeeded based on succeeded indexes.
+type SuccessPolicyCriterion struct {
+	// Specifies a set of required indexes when .spec.completionMode =
+	// "Indexed" in a text format.
+	// The job is declared as succeeded if a set of indexes are succeeded. 
+	// The list of indexes must come within 0 to `.spec.completions` and
+	// must not contain duplicates. At least one element is required. 
+	// The indexes are represented as decimal integers
+	// separated by commas. The numbers are listed in increasing order. Three or
+	// more consecutive numbers are compressed and represented by the first and
+	// last element of the series, separated by a hyphen.
+	// For example, if the completed indexes are 1, 3, 4, 5 and 7, they are
+	// represented as "1,3-5,7".
+	// 
+	// +optional
+	SucceededIndexes *string
+
+	// Specifies the required number of indexes when .spec.completionMode = "Indexed".
+	// Value can be an absolute number (ex: 5) when the job can be declared as succeeded.
+	//
+	// +optional
+	SucceededCount *int
+}
+...
+
+// These are valid conditions of a job.
+const (
+	// JobSuccessCriteriaMet means the job has been succeeded.
+	JobSucceessCriteriaMet JobConditionType = "SuccessCriteriaMet"
+	...
+)
+```
+
+Moreover, we validate the following constraints for the `criteria`:
+- whether the specified indexes in the `succeededIndexes` and
+  the number of indexes in the `succeededCount` don't exceed the value of `completions`.
+- whether `Indexed` is specified in the `completionMode` field.
+- whether the size of `succeededIndexes` is under 64Ki.
+- whether the `succeededIndexes` field has a valid format.
+- whether the `succeededCount` field has an absolute number.
+- whether the criteria haven't changed.
+
+In addition, when the `successPolicy` isn't nil, we validate the `criteria` isn't nil.
+
+### Evaluation
+
+The job-controller evaluates the successPolicies every in updating pod conditions in order,
+and when the job meets one of successPolicies, other successPolicies are ignored.
+After that, the job-controller adds a `SuccessCriteriaMet` condition instead of a `Failed` condition to `.status.conditions`.
+
+If a set of indexes are specified in the `succeededIndexes` field,
+the job-controller evaluates `.status.completedIndexes` to see if a set of indexes is there.
+
+If the number of indexes is specified in the `succeededCount`,
+the job-controller evaluates `.status.succeeded` to see if the value is `succeededCount` or more.
+
+In addition, the job controller performs actions against running pods based on the value of `whenCriterionAchieved`:
```

Comment:
Let's drop mentions of `whenCriterionAchieved` to Alternatives

Reply:
Done.